backend/src/core/services/credential.service.ts

```diff
@@ -292,6 +304,35 @@ const updateDemoDisplayed = async (
     isDisplayed: userDemo[0].isDisplayed,
   }
 }
+
+/**
+ * Updates the announcement version for the specified user
+ * @param userId
+ * @param announcementVersion
+ * @throws Error if user is not found
+ */
+const updateAnnouncementVersion = async (
+  userId: number,
+  announcementVersion: string
+): Promise<{ announcementVersion: string }> => {
+  const [rowUpserted] = await UserFeature.upsert(
+    { userId: userId, announcementVersion: announcementVersion },
+    { returning: true }
+  )
+
+  if (rowUpserted.userId !== userId) {
+    logger.error({
+      message: 'Updated incorrect user',
+      action: 'updateAnnouncementVersion',
+    })
+    throw new Error('Could not update announcement version')
+  }
```

Comment:
I believe it is very unlikely for the wrong userId to have been updated? Maybe instead you can check if `!rowUpserted` (is null) to ensure the upsert was successful.